torchtext/data/datasets_utils.py

```diff
@@ -172,8 +172,34 @@ def _dataset_docstring_header(fn, num_lines=None, num_classes=None):
         args_s += "\n     split: Only {default_split} is available."
         args_s += "\n         Default: {default_split}.format(default_split=default_split)"
 
+    example_item = _find_doc_example(fn.__name__, "train")
+    if example_item is not None:
+        args_s += "\nExamples:"
+        args_s += "\n    >>> train_iter = {}(split='train')".format(example_item["dataset_name"])
+        args_s += "\n    >>> next(train_iter)"
+        args_s += "\n    >>> ("
+        for item in example_item["first_line"]:
+            args_s += "\n    >>>        {}".format(_process_doc_example_items(item))
+        args_s += "\n    >>> )"
     return "\n".join([header_s, args_s]) + "\n"
 
+def _find_doc_example(dataset_name, split):
+    _PATH = download_from_url('https://raw.githubusercontent.com/pytorch/text/master/test/asset/raw_datasets.json')
```

Comment:
This adds an internet dependency at import time for each dataset. Instead let's store a json file with examples per dataset as needed and distribute together with the package, but let's limit it only for the first 50 characters.